deepchecks/vision/checks/performance/confusion_matrix.py

```diff
@@ -199,25 +211,178 @@ def update_object_detection(self, predictions, labels):
                     self.matrix['no-overlapping'][detected_class] += 1
 
-    def update_classification(self, predictions, labels):
+    def update_classification(self, batch: Batch):
         """Update the confusion matrix by batch for classification task."""
         assert self.matrix is not None
+        assert self.misclassified_images is not None
+
+        labels = batch.labels
+        predictions = batch.predictions
+        images = batch.images
 
-        for predicted_classes, image_labels in zip(predictions, labels):
+        for image, predicted_classes, image_labels in zip(images, predictions, labels):
             detected_class = max(range(len(predicted_classes)), key=predicted_classes.__getitem__)
             label_class = image_labels.item() if isinstance(image_labels, torch.Tensor) else image_labels
             self.matrix[label_class][detected_class] += 1
+            if label_class != detected_class:
+                self.misclassified_images.append((label_class, detected_class, image))
+
+    def _misclassified_images_thumbnails(
+        self,
+        matrix: pd.DataFrame,
+        dataset: VisionData
+    ) -> str:
+        assert self.misclassified_images is not None
+
+        if self.n_of_images_to_show == 0:
+            return ''
+
+        template = dedent(f"""
+            <h4>Misclassified Images</h4>
+            <p><b>NOTE:</b> in case of object detection task the next colors
+            are used for bbox identification:</p>
+            <ul>
+                <li>{self._LABEL_COLOR} - ground truth</li>
+                <li>{self._DETECTION_COLOR} - detected truth</li>
+            </ul>
+            {{note}}
+            {{grid}}
+        """)
+
+        misclassifications = (
+            (
+                label_class,
+                detected_class,
+                matrix.at[label_class, detected_class]  # count
+            )
+            for label_class, detected_class in product(list(matrix.index), repeat=2)
+            if label_class != detected_class
+        )
+        misclassifications = sorted(
+            misclassifications,
+            key=lambda it: it[2],
+            reverse=True
+        )
+        misclassified_images = [
+            (
+                (dataset.label_id_to_name(x_label), x_label),
+                (dataset.label_id_to_name(x_detected), x_detected),
+                img
+            )
+            for x_label, x_detected, _ in misclassifications
+            for (y_label, y_detected, img) in self.misclassified_images
+            if x_label == y_label and x_detected == y_detected
+        ]
+
+        if len(misclassified_images) == 0:
+            return ''
+
+        grid_content = [
+            '<span><b>Ground Truth</b></span>',
+            '<span><b>Detected Truth</b></span>',
+            '<span><b>Image</b></span>',
+        ]
+
+        for label, detected, img in misclassified_images[:self.n_of_images_to_show]:
```

Comment:
how do we know this won't be all the same "kind" misclassification? Anything here making sure we're gonna see different kinds of error (between different classes)?

Reply:
sorry, I did not understand what you mean.

>same "kind" misclassification

the same pair of classes? how is it possible?

Reply:
yeah, it can be that there are 100 images in which class 1 was mixed up with class 2. If they appear first, we'll get only them and miss out on examples of class 4 mixed up with class6 6. Can this happen here?

Reply:
oh, yeah, you are right.
I added `_IMAGES_PER_MISCLASSIFICATION` to fix this